src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusCollectionManager.cs

```diff
@@ -16,14 +16,16 @@ internal sealed class PrometheusCollectionManager
     private readonly Dictionary<Metric, PrometheusMetric> metricsCache;
     private readonly HashSet<string> scopes;
     private int metricsCacheCount;
-    private byte[] buffer = new byte[85000]; // encourage the object to live in LOH (large object heap)
+    private byte[] plainTextBuffer = new byte[85000]; // encourage the object to live in LOH (large object heap)
```

Comment:
@hez2010 - Curious if you explored the option of removing the cache instead of having two separate buffers?

Reply:
Not yet.
IMO there should only be one collector who collects those metrics at the same time. So, any incoming request will either triggering a new collect or waiting an existing collect, which results in some form of cache eventually.

Reply:
Thanks @hez2010 

> We may generate the view based on request type on demand instead of always generating two views for any type of request

Could you add a Todo comment for this, would be good to explore this as a follow up. The current solution is not ideal as it introduces a significant overhead of serializing the metrics in two different formats.